src/sentry/api/endpoints/organization_events_measurements_histogram.py

```diff
@@ -0,0 +1,55 @@
+from __future__ import absolute_import
+
+import sentry_sdk
+
+from rest_framework import serializers
+from rest_framework.response import Response
+
+from sentry.api.bases import OrganizationEventsV2EndpointBase, NoProjects
+from sentry.snuba import discover
+
+# The maximum number of measurements allowed to be queried at at time
+MAX_MEASUREMENTS = 4
+
+
+class MeasurementsHistogramSerializer(serializers.Serializer):
+    num_buckets = serializers.IntegerField(min_value=1)
+    precision = serializers.IntegerField(default=0, min_value=0, max_value=4)
+    measurement = serializers.ListField(allow_empty=False, max_length=4)
+
+
+class OrganizationEventsMeasurementsHistogramEndpoint(OrganizationEventsV2EndpointBase):
+    def get(self, request, organization):
+        if not self.has_feature(organization, request):
+            return Response(status=404)
+
+        try:
+            params = self.get_snuba_params(request, organization)
+        except NoProjects:
+            return Response([])
+
+        with sentry_sdk.start_span(
+            op="discover.endpoint", description="measurements_histogram"
+        ) as span:
+            span.set_tag("organization", organization)
+
+            serializer = MeasurementsHistogramSerializer(data=request.GET)
+            if serializer.is_valid():
+                data = serializer.validated_data
+
+                results = discover.measurements_histogram_query(
```

Comment:
is it worth wrapping this with a 
```suggestion
                with self.handle_query_errors():
                     results = discover.measurements_histogram_query(
```